modules/codegen/src/smithy4s/codegen/internals/Renderer.scala

```diff
@@ -811,17 +808,18 @@ private[internals] class Renderer(compilationUnit: CompilationUnit) { self =>
   }
 
   private def renderGetMessage(field: Field) = {
-    val hasModifier = !field.modifier.none
-    field match {
-      case field if field.tpe.isResolved && hasModifier =>
-        line"override def getMessage(): $string_ = ${field.name}"
-      case field if field.tpe.isResolved =>
-        line"override def getMessage(): $string_ = ${field.name}.orNull"
-      case field if hasModifier =>
-        line"override def getMessage(): $string_ = ${field.name}.value"
-      case field =>
-        line"override def getMessage(): $string_ = ${field.name}.map(_.value).orNull"
+    val fetchLogic = field.modifier.typeMod match {
```

Comment:
I still need to test that this works the way it should. (Even if part of me wants to say that anyone who annotates a field with `@errorMessage @nullable @required @default` or the like is getting what they deserve 🙈 )

Reply:
😆 yeah unfortunately this project is very much subject to Murphy's law, so I'm thankful with the additional patterns

Reply:
😂 OK, added another detailed smoke spec test that goes through the eight possible cases and makes sure it can retrieve the error message.